src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs

```diff
@@ -146,6 +146,57 @@ public static IDataView LoadFeaturizedAdultDataset(MLContext mlContext)
             return featurizedData;
         }
 
+        public static string DownloadMslrWeb10kTrain()
+        {
+            var fileName = "MSLRWeb10KTrain720kRows.tsv";
+            if (!File.Exists(fileName))
+                Download("https://tlcresources.blob.core.windows.net/datasets/MSLR-WEB10K/MSLR-WEB10K_Fold1.TRAIN.500MB_720k-rows.tsv", fileName);
+            return fileName;
+        }
+
+        public static string DownloadMslrWeb10kValidate()
+        {
+            var fileName = "MSLRWeb10KValidate240kRows.tsv";
+            if (!File.Exists(fileName))
+                Download("https://tlcresources.blob.core.windows.net/datasets/MSLR-WEB10K/MSLR-WEB10K_Fold1.VALIDATE.160MB_240k-rows.tsv", fileName);
+            return fileName;
+        }
+
+        public static (IDataView, IDataView) LoadFeaturizedMslrWeb10kTrainAndValidate(MLContext mlContext)
```

Comment:
>(IDataView, IDataView) [](start = 22, length = 22)

First this: [https://github.com/dotnet/machinelearning/issues/2501](https://github.com/dotnet/machinelearning/issues/2501)
Second, it's not validation data, it's test data, at least how you use it (you compute metrics with it, and don't pass it as validation dataset into trainer itself)
Third I think you can get away with one file instead of two. We have `TrainTestSplit` in ml.Ranking with `samplingKeyColumn` you can pass `GroupId` column.

can you modify this method to a) download one dataset, and return one dataview and in your examples use TrainTestSplit to break data into train and test data.